core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2674,16 +2730,24 @@ private void mountInternal(Supplier<JournalContext> journalContext, LockedInodeP
         if (UnderFileSystemUtils.isWeb(ufs)) {
           mountOption.setReadOnly(true);
         }
+      } catch (Throwable t) {
+        if (!ignoreError) {
+          throw t;
+        } else {
+          LogUtils.warnWithException(LOG, "Failed to acquire a UFS client for {}", ufsPath, t);
+        }
       }
-      // Check that the alluxioPath we're creating doesn't shadow a path in the parent UFS
-      MountTable.Resolution resolution = mMountTable.resolve(alluxioPath);
-      try (CloseableResource<UnderFileSystem> ufsResource =
-               resolution.acquireUfsResource()) {
-        String ufsResolvedPath = resolution.getUri().getPath();
-        if (ufsResource.get().exists(ufsResolvedPath)) {
-          throw new IOException(
-              ExceptionMessage.MOUNT_PATH_SHADOWS_PARENT_UFS.getMessage(alluxioPath,
-                  ufsResolvedPath));
+      if (!ignoreError) {
```

Comment:
Why do we allow this in the `ignoreError` case?

Reply:
Are you talking about the code from `2719` to `2739`? That is because the logic updates mount options based on a API call result from a UFS client, so skipping it will result in incorrect mount options.